osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -89,28 +103,108 @@ public void updateFirmware(final CorrelationIds ids,
         this.findOrganisation(ids.getOrganisationIdentification());
         final Device device = this.findActiveDevice(ids.getDeviceIdentification());
 
-        this.osgpCoreRequestMessageSender.send(
+        if (device instanceof Ssld) {
+            this.createSsldPendingFirmwareUpdateRecord(ids, firmwareUpdateMessageDataContainer.getFirmwareModuleData(),
+                    firmwareUpdateMessageDataContainer.getFirmwareUrl());
+        }
+
+        this.osgpCoreRequestMessageSender.sendWithScheduledTime(
                 new RequestMessage(ids,
                         this.domainCoreMapper.map(firmwareUpdateMessageDataContainer,
                                 org.opensmartgridplatform.dto.valueobjects.FirmwareUpdateMessageDataContainer.class)),
                 messageType, messagePriority, device.getIpAddress(), scheduleTime);
     }
 
+    private void createSsldPendingFirmwareUpdateRecord(final CorrelationIds ids,
+            final FirmwareModuleData firmwareModuleData, final String firmwareUrl) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "firmwareModuleData". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1172)